### PR TITLE
Add find and random logic for merchants

### PR DESCRIPTION
**Pull request by dannyradden**
* Add param finder to application controller
* Add sequence dates to merchants factory
* Add spec for find, find_all, and random
* Add routes and controllers for find and random in merchants namespace

@case-eee @somedayrainbows

### CODE REVIEW

**Comment by somedayrainbows** [comment]
Danny, this is possibly the best pull request I've ever seen. IN MY LIFE. 💅

File: app/controllers/api/v1/merchants/random_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::Merchants::RandomController < ApplicationController
+
+  def show
+    render json: Merchant.order("RANDOM()").first
```
Comment (case-eee) [inline app/controllers/api/v1/merchants/random_controller.rb]: this logic could be moved to a method on your `Merchant` model. the controller doesn't need to do the logic to find the correct merchant 👍

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::API
+  def param_finder
```
Comment (case-eee) [inline app/controllers/application_controller.rb]: is this needed in every single controller? if not, could we find another place to put this? we don't need every controller to know about this method if we can avoid it

File: spec/requests/api/v1/merchants_request_spec.rb [hunk 3]
```diff
@@ -57,4 +57,109 @@
     expect(invoices.count).to eq(invoices1.count)
     expect(invoices.first["status"]).to eq(invoices1.first.status)
   end
+
+  describe 'find endpoints'do
```
Comment (case-eee) [inline spec/requests/api/v1/merchants_request_spec.rb]: tests are looking great 🎉

**Pull request merged**